src/system_data.jl

```diff
@@ -314,12 +327,33 @@ function remove_components!(::Type{T}, data::SystemData) where {T}
     return remove_components!(T, data.components)
 end
 
+"""
+Removes the component from the main container and adds it to the masked container.
+"""
+function mask_component!(data::SystemData, component::InfrastructureSystemsComponent)
+    remove_component!(data.components, component, remove_time_series = false)
+    set_time_series_storage!(component, nothing)
+    add_masked_component!(
+        data,
+        component,
+        skip_validation = true,  # validation has already occurred
+        allow_existing_time_series = true,
+    )
+end
+
 function clear_time_series!(data::SystemData)
     clear_time_series!(data.time_series_storage)
     clear_time_series!(data.components)
     reset_info!(data.time_series_params)
 end
 
+function iterate_components_with_time_series(data::SystemData)
+    return Iterators.Flatten((
+        iterate_components_with_time_series(data.components),
+        iterate_components_with_time_series(data.masked_components),
```

Comment:
Note that we need to return time series of masked components in order to support `transform_single_time_series!` and `get_time_series_multiple`.